crates/typed-store/src/rocks/mod.rs

```diff
@@ -92,13 +92,144 @@ macro_rules! reopen {
+}
+
+macro_rules! delegate_call {
+    ($self:ident.$method:ident($($args:ident),*)) => {
+        match $self {
+            Self::DBWithThreadMode(d) => d.$method($($args),*),
+            Self::OptimisticTransactionDB(d) => d.$method($($args),*),
+        }
+    }
+}
+
+impl RocksDB {
+    pub fn get<K: AsRef<[u8]>>(&self, key: K) -> Result<Option<Vec<u8>>, rocksdb::Error> {
+        delegate_call!(self.get(key))
+    }
+
+    pub fn multi_get_cf<'a, 'b: 'a, K, I, W>(
+        &'a self,
+        keys: I,
+    ) -> Vec<Result<Option<Vec<u8>>, rocksdb::Error>>
+    where
+        K: AsRef<[u8]>,
+        I: IntoIterator<Item = (&'b W, K)>,
+        W: 'b + AsColumnFamilyRef,
+    {
+        delegate_call!(self.multi_get_cf(keys))
+    }
+
+    pub fn property_int_value_cf(
+        &self,
+        cf: &impl AsColumnFamilyRef,
+        name: impl CStrLike,
+    ) -> Result<Option<u64>, rocksdb::Error> {
+        delegate_call!(self.property_int_value_cf(cf, name))
+    }
+
+    pub fn get_pinned_cf<K: AsRef<[u8]>>(
+        &self,
+        cf: &impl AsColumnFamilyRef,
+        key: K,
+    ) -> Result<Option<rocksdb::DBPinnableSlice<'_>>, rocksdb::Error> {
+        delegate_call!(self.get_pinned_cf(cf, key))
+    }
+
+    pub fn cf_handle(&self, name: &str) -> Option<Arc<rocksdb::BoundColumnFamily<'_>>> {
+        delegate_call!(self.cf_handle(name))
+    }
+
+    pub fn create_cf<N: AsRef<str>>(
+        &self,
+        name: N,
+        opts: &rocksdb::Options,
+    ) -> Result<(), rocksdb::Error> {
+        delegate_call!(self.create_cf(name, opts))
+    }
+
+    pub fn drop_cf(&self, name: &str) -> Result<(), rocksdb::Error> {
+        delegate_call!(self.drop_cf(name))
+    }
+
+    pub fn delete_cf<K: AsRef<[u8]>>(
+        &self,
+        cf: &impl AsColumnFamilyRef,
+        key: K,
+    ) -> Result<(), rocksdb::Error> {
+        delegate_call!(self.delete_cf(cf, key))
+    }
+
+    pub fn path(&self) -> &Path {
+        delegate_call!(self.path())
+    }
+
+    pub fn put_cf<K, V>(
+        &self,
+        cf: &impl AsColumnFamilyRef,
+        key: K,
+        value: V,
+    ) -> Result<(), rocksdb::Error>
+    where
+        K: AsRef<[u8]>,
+        V: AsRef<[u8]>,
+    {
+        delegate_call!(self.put_cf(cf, key, value))
+    }
+
+    pub fn key_may_exist_cf<K: AsRef<[u8]>>(&self, cf: &impl AsColumnFamilyRef, key: K) -> bool {
+        delegate_call!(self.key_may_exist_cf(cf, key))
+    }
+
+    pub fn try_catch_up_with_primary(&self) -> Result<(), rocksdb::Error> {
+        delegate_call!(self.try_catch_up_with_primary())
+    }
+
+    pub fn write(&self, batch: rocksdb::WriteBatch) -> Result<(), TypedStoreError> {
+        match self {
+            RocksDB::DBWithThreadMode(db) => Ok(db.write(batch)?),
+            RocksDB::OptimisticTransactionDB(_) => Err(TypedStoreError::RocksDBError(
+                "operation not supported".to_string(),
+            )),
```

Comment:
We will most definitely need write with a write_batch to work for OptimisticTxDB. It may have its own `WriteBatch` implementation that we need to put under an enum

Reply:
sure, but on high level, I wanted to keep this PR reasonably small. 
Initially I was even thinking about just introducing enum and switching existing use cases to new enum.
So I wanted to ship this with MVP functionality and create a followup with some missing features.
Although some of them I want to discuss(understand better) cause it's not that trivial to add